### PR TITLE
[Iga] Differentiate integration edge vs coupling edge

### PR DESCRIPTION
**Pull request by tteschemacher**
Allows a differentiation within the IgaModeler between coupling edges and standard edges in the integration.

Adds the composite and coupling geometry type and family.

### CODE REVIEW

File: applications/IgaApplication/custom_modelers/iga_modeler.cpp [hunk 1]
```diff
@@ -83,8 +83,9 @@ namespace Kratos
         GetCadGeometryList(geometry_list, rCadModelPart, rParameters);
 
         if (!rParameters.Has("geometry_type")) {
+            std::string geometry_type = "";
```
Comment (manuelmessmer) [inline applications/IgaApplication/custom_modelers/iga_modeler.cpp]: Curly brackets? ;)

Comment (manuelmessmer) [reply applications/IgaApplication/custom_modelers/iga_modeler.cpp]: @KlausBSautter what do you think?

File: applications/IgaApplication/custom_modelers/iga_modeler.cpp [hunk 2]
```diff
@@ -156,8 +158,17 @@ namespace Kratos
                 }
             }
 
-            rGeometryList[i].CreateQuadraturePointGeometries(
-                geometries, shape_function_derivatives_order, integration_info);
+            if (rGeometryType == "SurfaceEdge"
```
Comment (manuelmessmer) [inline applications/IgaApplication/custom_modelers/iga_modeler.cpp]: couldn't you check this with the KRatosGeometryType?

**Review by manuelmessmer — DISMISSED** [review]
Looks goof to me.

File: applications/IgaApplication/custom_modelers/iga_modeler.cpp (hunk 2 above)
Comment (tteschemacher) [reply applications/IgaApplication/custom_modelers/iga_modeler.cpp]: Unfortunately, this doesn't work as those are actually different types. Those come from B-Rep terms and may not necessarily have a geometry counterpart in Kratos.

File: applications/IgaApplication/custom_modelers/iga_modeler.cpp (hunk 1 above)
Comment (tteschemacher) [reply applications/IgaApplication/custom_modelers/iga_modeler.cpp]: ;)

Comment (manuelmessmer) [reply applications/IgaApplication/custom_modelers/iga_modeler.cpp]: Much better now!

**Review by manuelmessmer — DISMISSED** [review]

**Review by rickyaristio — DISMISSED** [review]
Looks fine to me as well. Thanks, @tteschemacher!

**Review by rickyaristio — APPROVED** [review]

**Comment by tteschemacher** [comment]
thanks a lot for the review!

**Pull request merged**